src/compiler/crystal/syntax/ast.cr

```diff
@@ -1481,10 +1481,16 @@ module Crystal
     property type_vars : Array(ASTNode)
     property named_args : Array(NamedArgument)?
 
-    # `true` if this Generic was parsed from `T?`
-    property? question = false
+    property suffix : Suffix
 
-    def initialize(@name, @type_vars : Array, @named_args = nil)
+    enum Suffix
+      None
+      Question # T?
+      Asterisk # T*
+      Bracket  # T[N]
+    end
```

Comment:
I thought this change could also fix `foo : Foo****`, but I it doesn't. Do we need to know how many asterisks we have to fix this use case? Or this API is enough?

Reply:
Fixed this case in c918558. Thanks.